### PR TITLE
Improve the structs in the API, add docs

### PR DESCRIPTION
**Pull request by LVala**
This PR:
- removes most of the stuff that should be private from `RTPTransceiver`, `RTPSender` and `RTPReceiver` structs (like the "recorder..." modules)
- adds some getters to the `PeerConnection` (for the various states)
- adds doc stubs to most of the function in the API in general

### CODE REVIEW

File: lib/ex_webrtc/peer_connection.ex [hunk 1]
```diff
@@ -89,147 +118,292 @@ defmodule ExWebRTC.PeerConnection do
...
           RTPTransceiver.kind() | MediaStreamTrack.t(),
-          transceiver_options()
+          direction: RTPTransceiver.direction()
         ) :: {:ok, RTPTransceiver.t()}
   def add_transceiver(peer_connection, kind_or_track, options \\ []) do
     GenServer.call(peer_connection, {:add_transceiver, kind_or_track, options})
   end
 
+  @doc """
+  Sets the direction of transceiver specified by the `id`.
```
Comment (mickel8) [inline lib/ex_webrtc/peer_connection.ex:346]: Take a look at other functions too

```suggestion
  Sets the direction of transceiver specified by the `transceiver_id`.
```

File: lib/ex_webrtc/peer_connection.ex [hunk 2]
```diff
@@ -89,147 +118,292 @@ defmodule ExWebRTC.PeerConnection do
...
+  def remove_track(peer_connection, sender_id) do
+    GenServer.call(peer_connection, {:remove_track, sender_id})
   end
 
   @doc """
-  Returns peer connection's statistics.
+  Closes the PeerConnection.
 
-  See [RTCStatsReport](https://www.w3.org/TR/webrtc/#rtcstatsreport-object) for the output structure.
+  This function kills the PeerConnection process.
```
Comment (mickel8) [inline lib/ex_webrtc/peer_connection.ex:404]:
```suggestion
  This function kills the `ExWebRTC.PeerConnection` process.
```

File: lib/ex_webrtc/rtp_receiver/report_recorder.ex [hunk 3]
```diff
@@ -39,10 +39,7 @@ defmodule ExWebRTC.RTPReceiver.ReportRecorder do
             jitter: 0.0,
             total_lost: 0
 
-  @doc """
-  Records incoming RTP Packet.
-  `time` parameter accepts output of `System.monotonic_time()` as a value.
-  """
+  # `time` parameter accepts output of `System.monotonic_time()` as a value.
```
Comment (mickel8) [inline lib/ex_webrtc/rtp_receiver/report_recorder.ex:42]: IMO the previous docs could stay

File: lib/ex_webrtc/rtp_sender/report_recorder.ex [hunk 4]
```diff
@@ -80,13 +77,9 @@ defmodule ExWebRTC.RTPSender.ReportRecorder do
...
   end
 
-  @doc """
-  Creates an RTCP Sender Report.
-  `time` parameter accepts output of `System.os_time(:native)` as a value (UNIX timestamp in :native units).
-
-  This function can be called only if at least one packet has been recorded,
-  otherwise it will raise.
-  """
+  # `time` parameter accepts output of `System.os_time(:native)` as a value (UNIX timestamp in :native units).
```
Comment (mickel8) [inline lib/ex_webrtc/rtp_sender/report_recorder.ex:80]: As above

**Review by mickel8 — APPROVED** [review]
Coool :+1:

File: lib/ex_webrtc/rtp_receiver/report_recorder.ex (hunk 3 above)
Comment (LVala) [reply lib/ex_webrtc/rtp_receiver/report_recorder.ex:42]: I'd say to leave as a comment and in general use comments for developer information and docs for public stuff? The only useful thing I could think off was that the LSP would provide the doc info even when function is private, but it doesn't seem to be the case.

**Pull request merged**